src/Network/Algorand/Node/Api.hs

```diff
@@ -117,13 +123,70 @@ newtype TransactionsRep = TransactionsRep
   } deriving (Generic, Show)
 $(deriveJSON algorandCamelOptions 'TransactionsRep)
 
+data TealValue = TealValue
+  { tvBytes :: ByteString
+  , tvUint :: Word64
+  , tvType :: Word64
+  } deriving stock Show
+$(deriveJSON algorandTrainOptions 'TealValue)
+
+tealValueBytesType :: Word64
+tealValueBytesType = 1
+
+tealValueUintType :: Word64
+tealValueUintType = 2
+
+data TealKvEntry = TealKvEntry
+  { tkeKey :: ByteString
+  , tkeValue :: TealValue
+  } deriving stock Show
+$(deriveJSON algorandTrainOptions 'TealKvEntry)
+
+type TealKvStore = [TealKvEntry]
+
+data Asset = Asset
+  { asAmount :: Word64
+  -- ^ Number of units held.
+  , asAssetId :: AssetIndex
+  -- ^ Asset ID of the holding.
+  , asCreator :: Address
+  -- ^ Address that created this asset.
+  -- This is the address where the parameters for this asset can be found, and
+  -- also the address where unwanted asset units can be sent in the worst case.
+  , asDeleted :: Maybe Bool
+  -- ^ Whether or not the asset holding is currently deleted from its account.
+  , asIsFrozen :: Bool
+  -- ^ Whether or not the holding is frozen.
+  , asOptedInAtRound :: Maybe Round
+  -- ^ Round during which the account opted into this asset holding.
+  , asOptedOutAtRound :: Maybe Round
```

Comment:
It looks a bit sad when all fields are commented except for one.

Reply:
ops.. according to source code, some of this fields not presented any more